lib/runners/processor/clang_tidy.rb

```diff
@@ -0,0 +1,99 @@
+module Runners
+  class Processor::ClangTidy < Processor
+    Schema = StrongJSON.new do
+      let :runner_config, Schema::BaseConfig.base.update_fields { |fields|
+        fields.merge!({
+          apt: enum?(string, array(string)),
+          'include-path': enum?(string, array(string))
+        })
+      }
+
+      let :issue, object(
+        severity: string,
+      )
+    end
+
+    register_config_schema(name: :clang_tidy, schema: Schema.runner_config)
+
+    # @see https://github.com/github/linguist/blob/775b07d40c04ef6e0051a541886a8f1e30a892f4/lib/linguist/languages.yml#L532-L535
+    # @see https://github.com/github/linguist/blob/775b07d40c04ef6e0051a541886a8f1e30a892f4/lib/linguist/languages.yml#L568-L584
+    VALID_EXTENSIONS = [".c", ".cpp", ".c++", ".cc", ".cp", ".cxx"].freeze
+    GLOB_HEADERS = "**/*.{h,h++,hh,hpp,hxx,inc,inl,ipp,tcc,tpp}".freeze
+
+    def setup
+      deploy_packages
+      yield
+    end
+
+    def analyzer_bin
+      "clang-tidy"
+    end
+
+    def analyze(changes)
+      issues = []
+
+      changes
+        .changed_paths
+        .select { |path| VALID_EXTENSIONS.include?(path.extname.downcase) }
+        .map{ |path| relative_path(working_dir / path, from: current_dir) }
+        .reject { |path| path.to_s.start_with?("../") } # reject files outside the current_dir
```

Comment:
[suggestion] Maybe the following code seems more safe and clear to me:

```suggestion
        .reject { |path| path.parent.to_path == "..") } # reject files outside the current_dir
```

Reply:
Let's take an example below:

```
/tmp/blablabla/WORKING_DIR/CURRENT_DIR/foo/bar.c
                          /aaa/bbb/ccc.c
```

We have `../aaa/bbb/ccc.c` (and `foo/bar.c`). Apply it to the `path.parent.to_path`, then we get "../aaa/bbb" which is not equal to "..".
Alternatively, we can use `path.parent.to_path.split(File::SEPARATOR)[0] == ".."`, but it is a bit complicated.

Reply:
Make sense. 👍